macrosynergy/pnl/proxy_pnl_calc.py

```diff
@@ -107,3 +250,45 @@ def proxy_pnl_calc(
         if not isinstance(_varx, _typex):
             raise TypeError(f"{_namex} must be {_typex}")
 
+        if _typex in [list, str, dict] and len(_varx) == 0:
+            raise ValueError(f"`{_namex}` must not be an empty {str(_typex)}")
+
+    if start is None:
+        start = df["real_date"].min().strftime("%Y-%m-%d")
+    if end is None:
+        end = df["real_date"].max().strftime("%Y-%m-%d")
+
+    # Reduce the dataframe - keep only the txn costs, and the spos xcats
+    df = reduce_df(
+        df=df,
+        start=start,
+        end=end,
+        blacklist=blacklist,
+    )
+
+    # Initialize the TransactionCosts class
+    transcation_costs: TransactionCosts = TransactionCosts(
+        df=df,
+        fids=fids,
+        tcost_n=tcost_n,
+        rcost_n=rcost_n,
+        size_n=size_n,
+        tcost_l=tcost_l,
+        rcost_l=rcost_l,
+        size_l=size_l,
+    )
+
+    # Create the wide dataframe
+    df_wide = qdf_to_ticker_df(df=df)
+
+
+if __name__ == "__main__":
+    ...
+    dfxcn = pd.read_pickle("data/dfxcn.pkl")
```

Comment:
Should we leave pickled data? If we are adding data to it, I would propose at least as a `CSV` so it is humanly readable. It is a very small dataset.

Reply:
I don't think so... I'm currently using it so I can "hand-off" from the notebooks